src/app/scripts/cac/pages/cac-pages-home.js

```diff
@@ -305,12 +305,12 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, TripOptions, SearchPara
     // Note that components are responsible for doing the right thing based on whether they're
     // active or not.
     function onUrlChanged() {
-        setActiveTab();
         modeOptionsControl.setMode(UserPreferences.getPreference('mode'));
-        directionsControl.setFromUserPreferences();
-        exploreControl.setFromUserPreferences();
         directionsFormControl.setFromUserPreferences();
         showHideNeedWheelsBanner();
+        directionsControl.setFromUserPreferences();
+        exploreControl.setFromUserPreferences();
+        setActiveTab();
```

Comment:
I think there's a catch-22 here, or at least a bit of a tangled web.  I may have been wrong in moving toward pushing "do the right thing" logic out of Home and into the controllers.  With `setActiveTab()` last I'm seeing a bug whereby using browser navigation to go from Explore to Home results in the URL getting rewritten to the Explore URL because `exploreControl.setFromUserPreferences();` gets triggered while Explore is still active, so it calls `clickedExplore`, which runs (debounced) and updates the URL to match the parameters it's running with.

It's the same series of events I was talking about [here](https://github.com/azavea/cac-tripplanner/pull/719#discussion_r93785240), but I didn't realize that superfluous calls on exiting tabs would have visible negative effects.  Here's one possibility, which seems inelegant but is easier than rethinking the whole setup:

```
diff --git a/src/app/scripts/cac/control/cac-control-tab.js b/src/app/scripts/cac/control/cac-control-tab.js
index a1db9c8..77c2834 100644
--- a/src/app/scripts/cac/control/cac-control-tab.js
+++ b/src/app/scripts/cac/control/cac-control-tab.js
@@ -79,6 +79,9 @@ CAC.Control.Tab = (function ($) {
             }
 
             this.events.trigger(eventNames.tabShown, tabId);
+            return true;
+        } else {
+            return false;
         }
     }
 
diff --git a/src/app/scripts/cac/pages/cac-pages-home.js b/src/app/scripts/cac/pages/cac-pages-home.js
index d6ee99b..dba61e0 100644
--- a/src/app/scripts/cac/pages/cac-pages-home.js
+++ b/src/app/scripts/cac/pages/cac-pages-home.js
@@ -224,12 +224,12 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, TripOptions, SearchPara
         if (!UserPreferences.isDefault('method')) {
             var method = UserPreferences.getPreference('method');
             if (method === 'directions') {
-                tabControl.setTab(tabControl.TABS.DIRECTIONS);
+                return tabControl.setTab(tabControl.TABS.DIRECTIONS);
             } else if (method === 'explore') {
-                tabControl.setTab(tabControl.TABS.EXPLORE);
+                return tabControl.setTab(tabControl.TABS.EXPLORE);
             }
         } else {
-            tabControl.setTab(tabControl.TABS.HOME);
+            return tabControl.setTab(tabControl.TABS.HOME);
         }
     }
 
@@ -308,9 +308,11 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, TripOptions, SearchPara
         modeOptionsControl.setMode(UserPreferences.getPreference('mode'));
         directionsFormControl.setFromUserPreferences();
         showHideNeedWheelsBanner();
-        directionsControl.setFromUserPreferences();
-        exploreControl.setFromUserPreferences();
-        setActiveTab();
+        var tabChanged = setActiveTab();
+        if (!tabChanged) {
+            directionsControl.setFromUserPreferences();
+            exploreControl.setFromUserPreferences();
+        }
     }
 
     function closedTripModal(event) {
```

I.e. since `onTabSelected` methods run `setFromUserPreferences` when the tab is entering, only manually run it when the URL change doesn't result in a tab change.  Or it might make sense to only run the `setFromUserPreferences` for the active tab.